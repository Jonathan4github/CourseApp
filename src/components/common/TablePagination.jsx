import React from 'react';
import PropTypes from 'prop-types';

import ReactPaginate from 'react-paginate';

const propTypes = {
  pageCount: PropTypes.number.isRequired,
  currentPage: PropTypes.number.isRequired,
  handlePageChange: PropTypes.func.isRequired
};

const TablePagination = ({ pageCount, currentPage, handlePageChange }) => (
  <div className="text-center">
    <ReactPaginate
      breakLabel="..."
      breakClassName="break-me"
      pageCount={pageCount}
      initialPage={currentPage - 1}
      marginPagesDisplayed={2}
      pageRangeDisplayed={3}
      onPageChange={handlePageChange}
      disableInitialCallback
      containerClassName="pagination"
      subContainerClassName="pages pagination"
      activeClassName="active"
    />
  </div>
);

TablePagination.propTypes = propTypes;

export default TablePagination;
