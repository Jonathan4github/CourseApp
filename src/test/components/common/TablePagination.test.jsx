/* eslint-disable */
import TablePagination from '../../../components/common/TablePagination';

const setup = (pageCount) => {
  const props = {
    pageCount,
    currentPage: 1,
    handlePageChange: jest.fn()
  };

  return mount(<TablePagination {...props} />);
};

let mountWrapper;

describe('<TablePagination /> component test', () => {
  it('renders component without crashing', () => {
    mountWrapper = setup(1);
    expect(toJson(mountWrapper)).toMatchSnapshot();
  });

  it('renders appropriate elements', () => {
    mountWrapper = setup(1);
    expect(mountWrapper.find('.pagination-container').length).toEqual(1);
  });
});
