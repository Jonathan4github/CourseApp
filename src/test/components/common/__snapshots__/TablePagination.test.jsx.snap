// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TablePagination /> component test renders component without crashing 1`] = `
<TablePagination
  currentPage={1}
  handlePageChange={[MockFunction]}
  pageCount={1}
>
  <div
    className="pagination-container"
  >
    <PaginationBoxView
      activeClassName="active"
      breakClassName="break-me"
      breakLabel="..."
      containerClassName="pagination"
      disableInitialCallback={true}
      disabledClassName="disabled"
      initialPage={0}
      marginPagesDisplayed={2}
      nextClassName="next"
      nextLabel="Next"
      onPageChange={[MockFunction]}
      pageCount={1}
      pageRangeDisplayed={3}
      previousClassName="previous"
      previousLabel="Previous"
      subContainerClassName="pages pagination"
    >
      <ul
        className="pagination"
      >
        <li
          className="previous disabled"
        >
          <a
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
          >
            Previous
          </a>
        </li>
        <PageView
          activeClassName="active"
          key="0"
          onClick={[Function]}
          page={1}
          selected={true}
        >
          <li
            className="active"
          >
            <a
              aria-current="page"
              aria-label="Page 1 is your current page"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex="0"
            >
              1
            </a>
          </li>
        </PageView>
        <li
          className="next disabled"
        >
          <a
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
          >
            Next
          </a>
        </li>
      </ul>
    </PaginationBoxView>
  </div>
</TablePagination>
`;
